alm/src/utils/web3.ts

```diff
@@ -61,3 +62,11 @@ export const getBlock = async (web3: Web3, blockNumber: number): Promise<BlockTr
     }
     return result
   })
+
+export const getChainId = async (web3: Web3, snapshotProvider: SnapshotProvider) => {
+  let id = snapshotProvider.chainId()
```

Comment:
We don't get lots of advantages if get the chain id from the snapshot rather than from the chain. At the same time, if something goes wrong (e.g. a chain with an another chain id is available through JSON-RPC URL) getting a real value from the chain will not confuse users.

Instead I suggest to use some info (e.g. the block hash of the latest block) from the snapshot to confirm that the snapshot was taken for the same chain as is being used in the ALM app when a user works with it.

Reply:
I'm not sure I fully understood your suggestion and what are we trying to achieve with it.
The advantage of getting the chainId from the snapshot is to avoid doing 2 requests to get the chainId from the chain (one for home and one for foreign).

The snapshot script creates two files, one for home network (in `src/snapshots/home.json`) and one for foreign network (in `src/snapshots/foreign.json`) with the information collected in the latests available block using the env vars for the RPC URLS and the bridge contract addresses. So all the information stored in the snapshot will be related to the same chain.

So when the user visits a page, the chainId will be read from the snapshot instead of sending a request to the RPC URL. If the RPC URL is wrong and now it is providing data from a different chain most probably the information fetched from the RPC will be all wrong, the address of the bridge contract provided by env var could not exist in that chain, it won't be able to get the receipt, the correct addresses, list of validators, etc, even if the snapshot is not present.

Reply:
OK. I see you point. The contract addresses will be mostly different for another chain so most probably the ALM will not work at all.